core/src/main/java/org/pentaho/di/connections/vfs/VFSConnectionManagerWrapper.java

```diff
@@ -0,0 +1,115 @@
+package org.pentaho.di.connections.vfs;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import org.pentaho.di.connections.ConnectionDetails;
+import org.pentaho.di.connections.ConnectionManager;
+import org.pentaho.di.connections.ConnectionProvider;
+import org.pentaho.di.connections.vfs.provider.ConnectionFileName;
+import org.pentaho.di.core.bowl.Bowl;
+import org.pentaho.di.core.exception.KettleException;
+import org.pentaho.di.core.vfs.IKettleVFS;
+
+import java.util.List;
+
+/**
+ * The {@code VFSConnectionManagerWrapper} interface contains functionality related with VFS connections.
+ */
+public interface VFSConnectionManagerWrapper {
```

Comment:
This interface and accompanying implementation:
1. represent a specific connection manager from a bowl
2. include a pointer back to the bowl, as this is frequently needed; note that bowls have more than one connection manager (RO/RW, explicit); this allows manager code to access other bowl-specific objects, such as `IKettleVFS`, without having to receive the bowl the manager was resolved from on every method.
3. a VFS only view of corresponding `ConnectionManager` methods (e.g. `getNames()`, `getTypes`), directly filtering out non-`VFSConnectionProvider`/`VFSConnectionDetails` instances.
4. includes logic that would belong to the manager (given it is not provider-specific) but which is VFS-specific.

Reply:
@abryant-hv in https://github.com/pentaho/pentaho-kettle/pull/9334#pullrequestreview-2071681357
> The new Wrapper seems OK, even though it's only used in a couple places.


As described above, although the new wrapper class offers some sugar for handling VFS connections, it also provides exclusive VFS-specific functionality.

The wrapper name may be misleading. It intends to reflect that it is an associated and secondary type with/to the normal `ConnectionManager`.

It's not used from many places, atm. Only from the new code being proposed. But could be used elsewhere, e.g. `VFSFileProvider`. Premature atm, imo, but I believe it would enable getting rid of some of the boilerplate out there :-)

Reply:
Ended up following a different approach.
Added some sugar methods to `ConnectionDetails`.
Then created a `VFSConnectionManagerHelper` class.
It is used by `ConnectionManager`, to delegate VFS functionality to. For example, `ConnectionManager#test(ConnectionDetails)` knows to dynamically delegate to the VFS helper for `VFSConnectionDetails`.
It is also used directly by other classes, such as the file name transformers, for VFS functionality not exposed through the connection manager.

Another thing which resulted in simpler signatures overall, was the addition of `ConnectionManager#getBowl()`, https://github.com/pentaho/pentaho-kettle/pull/9390. This was one of the reasons for which the wrapper existed!